### PR TITLE
[flutter_tools] Reland flutter assemble for iOS

### LINKED ISSUE #32925: Re-implement existing iOS and Android build in terms of --legacy flutter assemble targets
This should allow us to re-enable build caching, though this may also require work from gradle.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Updates xcode_backend.sh to use (mostly) assemble APIs. This almost completes the migration, though I still need to move the asset unpack step - this is slightly more complicated due to the module format.

Should have no new external behavior, but does allow us to more easily wire up new features like font-subset, dart-defines, and obfuscation (which was never implemented for iOS?)

FYI @dnfield

Fixes #32925

Reland: Makes the Universal framework depend on kernel. Due to the behavior of the codegen integration, running them at the same time causes parts of .dart_tool to be deleted and recreated.

Reland: Don't specify an armv7 (32 bit) arch unless it is listed as active arches, since this isn't supported on newer deployment targets

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I also need to update the expected analytics for the module_test_ios

File: packages/flutter_tools/lib/src/build_system/targets/ios.dart [hunk 1]
```diff
@@ -176,12 +176,19 @@ class DebugUniveralFramework extends Target {
...
+      ?.map(getIOSArchForName)
+      ?.toSet()
+      ?? <DarwinArch>{DarwinArch.arm64};
     final File iphoneFile = environment.buildDir.childFile('iphone_framework');
     final File simulatorFile = environment.buildDir.childFile('simulator_framework');
     final File lipoOutputFile = environment.buildDir.childFile('App');
     final RunResult iphoneResult = await createStubAppFramework(
       iphoneFile,
       SdkType.iPhone,
+      // Only include 32bit if it is contained in the active arches.
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/build_system/targets/ios.dart]: "active architectures"

**Review by jmagman — APPROVED** [review]
module_test_ios passes 🎲 🎲 🎲

**Comment by jonahwilliams** [comment]
Since the last commit passed, and these checks are obviously flakes I'm going to submit anyway

**Pull request merged**